src/native/libs/System.Security.Cryptography.Native/pal_evp_pkey.c

```diff
@@ -89,7 +89,220 @@ int32_t CryptoNative_UpRefEvpPkey(EVP_PKEY* pkey)
     return EVP_PKEY_up_ref(pkey);
 }
 
-static bool CheckKey(EVP_PKEY* key, int32_t algId, int32_t (*check_func)(EVP_PKEY_CTX*))
+static bool Lcm(const BIGNUM* num1, const BIGNUM* num2, BN_CTX* ctx, BIGNUM* result)
+{
+    assert(result);
+
+    // lcm(num1, num2) = (num1 * num2) / gcd(num1, num2)
+    BIGNUM* mul = NULL;
+    BIGNUM* gcd = NULL;
+    bool ret = false;
+
+    if ((mul = BN_new()) == NULL ||
+        (gcd = BN_new()) == NULL ||
+        !BN_mul(mul, num1, num2, ctx) ||
```

Comment:
In general context, it's probably faster to do `LCM(a, b) = a/GCD(a, b) * b/GCD(a, b)`, to reduce the BN_size for `a*b`.  But, if this is just for LCM(p-1, q-1); I believe the answer for GCD is basically 2, meaning the predivision won't help all that much (especially compared against the storage and cleanup of the BN values to store the division results).

But, if you want to explore that avenue, I'd be curious which way it works out :smile:.

Reply:
Internal conversation with myself:

That isn't the same thing. Let's say a is 5 and b is 10. We know by instinct the LCM here is 10. The GCD is 5. Using Jeremy's suggestion, (5 / 5) * (10 / 5) that's 2.

*thinking, thinking, thinking*

Oh, he probably meant `a/GCD(a, b) * b`

At this point, it's the same operations in a different order. Benchmark shows no real difference between the two.

Reply:
Oh, yeah.  Something did feel off when I was writing it.

It's a good thing my math degree is only in discrete mathematics, which doesn't apply to integer math... oh, wait, that's only exactly what it is in.

I blame a lack of coffee :smile: